### ISSUE TITLE
dataset generation: each training dataset with "uniform map"?

### ISSUE THREAD

**Issue by AceCoooool** [cszn/FFDNet#3]
Each training data with an uniform map?  (or is there some training data with spatially variant noise map in training stage?)

thank you :smile:

**Comment by cszn** [comment]
I used uniform map.

**Comment by AceCoooool** [comment]
Thank you kai :+1: 

Sorry to ask another question: (I want to achieve the training part)
The loss function: (y_pred-y)^2/(noise_level)   or   [(y_pred-y)/(noise_level)]^2  (through my previous experiment in DnCNN-B, the first one can get similar results in your paper.)

**Comment by cszn** [comment]
I think pixel-level loss functions such as L1 and L2 can give the same final results.
I used [this loss function.](https://github.com/cszn/DnCNN/blob/master/TrainingCodes/DnCNN_TrainingCodes_v1.1/vl_nnloss.m)

**Comment by AceCoooool** [comment]
Thank you .
However, If the batch-size is small. there is a considerable oscillation in loss. (Due to each batch's noise level is far away. ) Yeah, large-batch without this problem.~ Thank you again~ :smile:
